src/Snippets/Docs/Testing.cs

```diff
@@ -58,4 +62,68 @@ public static void GetPipelineDescriptorGeneric()
 
         #endregion
     }
+
+    public static void PipelineProviderProviderMocking()
+    {
+        #region testing-resilience-pipeline-provider-mocking
+
+        ResiliencePipelineProvider<string> pipelineProvider = Substitute.For<ResiliencePipelineProvider<string>>();
+
+        // Mock the pipeline provider to return an empty pipeline for testing
+        pipelineProvider
+            .GetPipeline("my-pipeline")
+            .Returns(ResiliencePipeline.Empty);
+
+        // Use the mocked pipeline provider in your code
+        var api = new MyApi(pipelineProvider);
+
+        // You can now test the api
+
+        #endregion
+    }
+}
+
+#region testing-resilience-pipeline-provider-usage
+
+// Represents an arbitrary API that needs resilience support
+public class MyApi
+{
+    private readonly ResiliencePipeline _pipeline;
+
+    // The pipelineProvider is injected via dependency injection
+    public MyApi(ResiliencePipelineProvider<string> pipelineProvider)
+    {
+        _pipeline = pipelineProvider.GetPipeline("my-pipeline");
+    }
+
+    public async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        await _pipeline.ExecuteAsync(
+            static async token =>
+            {
+                // Add your code here
+            },
+            cancellationToken);
+    }
 }
+
+// Extensions to incorporate MyApi into dependency injection
+public static class MyApiExtensions
+{
+    public static IServiceCollection AddMyApi(this IServiceCollection services)
+    {
+        services.AddResiliencePipeline("my-pipeline", builder =>
+        {
+            builder.AddRetry(new RetryStrategyOptions
+            {
+                MaxRetryAttempts = 4
+            });
+        });
+
+        services.TryAddSingleton<MyApi>();
```

Comment:
For this simple example I don't think we need to use the Try variant.

```suggestion
        services.AddSingleton<MyApi>();
```